collector/src/lib.rs

```diff
@@ -15,24 +15,33 @@ pub use self_profile::{QueryData, SelfProfile};
 #[derive(Debug, Copy, Clone, PartialEq, PartialOrd, Deserialize)]
 pub struct DeltaTime(#[serde(with = "round_float")] pub f64);
 
+/// The bound of a range changes in codebase
```

Comment:
`Bound` doesn't seem to always be a bound in the true sense. If it's a `Commit` we're usually looking for exact matches. I'm not sure what `Bound` is really supposed to be.

Reply:
It is supposed to be "exact" in the sense of identifying a particular commit in our storage, at least at the time of access.

Commits hashes are easy, obviously, though the string in Commit(String) can also be the artifact ID (e.g., 1.53.0 or whatever the user provides on the command line when recording data locally).